### ISSUE TITLE
Systems outfit list

### ISSUE THREAD

**Issue by Nescio0** [endless-sky/endless-sky#3375]
Currently the "Systems" category includes all outfits which do not fit in any other category, which is perfectly fine for miscellaneous things such as bunks, cargo expansions, fuel pods, outfits expansions, quantum key stones, and the cloaking device.

However, the (seventeen?) shield generator and hull repair systems form a coherent group with a dedicated function, therefore it would make sense to list them separately under "Regenerators".
(Not "Generators", because they do not actually generate anything; your ship already as shields and hull without those; they mearly restore health. Furthermore, generator is probably better applicable to energy generators, e.g. "Generator (Furnace Class)".)

Likewise, the (eleven?) cooling systems could be grouped together under "Cooling", since these all reduce heat. (For comparison, energy providers are grouped under "Power".)

The advantages of this are to make it easier to compare similar items and also to be able to find more quickly what you're looking for under "Systems".

**Comment by gunqqer** [comment]
I am under the mindset that we need subcategories, so systems/cooling or systems/regenerators (that's supposed to mimic file system architecture) Or just different sorting options like by cost or certain attributes they have

**Comment by Amazinite** [comment]
Essentially a duplicate of #2166, but with a different suggestion of exactly how systems should be broken down more.

**Comment by Bladewood** [comment]
The thing is, these kinds of outfits aren't as [essential, fundamental, special] as the existing ones.
Engines are *required* for a ship to get anywhere.
Weapons are *required* for a fundamental game mechanic: blowing up other ships.
Power outfits (including both generators and batteries) are *necessary* to drive engines and weapons, and are very important when it comes to the concept of tier progression.

You can get by without regeneration outfits.
You can get by without cooling, assuming you're being conservative with your heat output.

**Comment by jafdy** [comment]
While I agree that it would be good to split down Systems, no one has yet found a good way of doing it. Most either do go far enough and just feel . . . lacking, just not quite enough to be worth making the change. Or goes to far and we get so many categories that some of them are just too small.

**Comment by Nescio0** [comment]
#2166 suggests to introduce sub-categories, which might be interesting, but seems to me rather arbitrary, because many of the current categories might as well be sub-categories, and vice versa.

What I suggested in the first post, the introduction of one or two new coherent categories of about a dozen outfits, is basically conservative, because it is in line with the current category logic structure. 

> You can get by without regeneration outfits.

Likewise, you can get by without hand to hand weapons, or turrets, or any other weapons, yet there are categories for those. 

> Or goes to far and we get so many categories that some of them are just too small.

There are four ramscoops, which form a coherent group, however, proposing to create a separate category for those goes probably too far, I fully agree.


To clarify, I'm not claiming my proposal is the perfect solution. However, I do consider it an improvement to the current situation. I'm open to counter-proposals etc., of course.

**Comment by jafdy** [comment]
Remember, if we add a category we can never remove that category.

**Comment by Wrzlprnft** [comment]
One thing is: Yes, there are many Outfits in the System section, if you have all of them on a planet. but they are quite spread out, so the section isn't always that convoluted.

**Comment by gunqqer** [comment]
Not everything needs a catagory. Until 5, maybe even 10 it can go into say, "misc"

**Comment by Nescio0** [comment]
> One thing is: Yes, there are many Outfits in the System section, if you have all of them on a planet. but they are quite spread out, so the section isn't always that convoluted.

If I want to figure out which shield system I need for a particular ships (and where it's sold), I open the map, click on the Outfitters tab, and start using that one-column outfits list to compare items, which requires a lot of scrolling up and down. Not having dozens of unrelated systems in between would make it a lot less annoying to compare e.g. the D14-RN and the Small Shield Module.
I'm not the only one who uses the map to compare available outfits, am I?

> Not everything needs a catagory. Until 5, maybe even 10 it can go into say, "misc"

I fully agree. (There are more shielding and cooling systems, though.)

**Comment by tehhowch** [comment]
Press Shift to compare when clicking outfits.

**Comment by Nescio0** [comment]
> Press Shift to compare when clicking outfits (or ships).

That works only for a pair. If I'm looking for a mass<20 shielding, I have four available options (Bright Could, D14-RN, Hai Corundum, Small Shield Module). Besides, the Shift feature does not reduce the amount of back and forth scrolling required to go from one to another.

**Comment by tehhowch** [comment]
You mentioned `one-column outfits list`, `compare items`, `a lot of scrolling up and down`, and named two items. It's certainly appropriate to assume that you are unaware of the shift-click compare functionality given those conditions.

A better comparison UI (namely, not a single-column list and using filters as opposed to collapsing categories) would be a massive improvement and obviate the need for new categories, as sensible design of the player experience precludes offering every technology at the same location at the same time.

**Comment by Nescio0** [comment]
> A better comparison UI (namely, not a single-column list and using filters as opposed to collapsing categories) would be a massive improvement

Certainly. However, even if you could do that neatly in the forseeable future, separating one or two new categories from the very large Systems grouping could still be an improvement.

**Comment by tehhowch** [comment]
I'm imaging the filters completely resolving that need. They'd be populated based on the accumulation of all known outfits' attributes, and binned appropriately, so you'd be able to select outfits with `Shield Regeneration > 0` and `Mass [10, 20]`

**Comment by Nescio0** [comment]
That would mean creating many filters. If there are filters for every possible outfit variable, then do we still need any categories anyway? It's certainly an interesting alternative.

**Comment by gunqqer** [comment]
Filters would easily solve this. We keep the existing catagories but then have auto-filters to easily sort by X or Y, or both

**Comment by tehhowch** [comment]
As an interim step to fully dynamic filters based on any outfit attribute known to the player, we can hard-code methods to interact with basic "priority" filters like energy generation, thrust, turn, outfit space, shield gen, hull repair, and cooling. Then when someone implements a sensible method of alternative filter input, we can open up to the full gamut.

I'm busy with other projects for the game, so feel free to start developing this UI. Save the spritework for the end.

**Comment by endless-sky** [comment]
The current number of items in each category is:

category | count
---- | ----
Guns | 26
Turrets | 40
Secondary Weapons | 15
Ammunition | 24
Systems | 59
Power | 57
Engines | 86
Hand to Hand | 6
Special | 15

Systems are a relatively large category, but not the largest. And, if you split out shield / hull generators and cooling, those would become some of the smallest categories (not counting the special two at the end).

There are very few planets where the outfitter sells so many Systems that the category is hard to read through. And if those categories were split out, in many outfitters there would only be one or two outfits in each of those categories, leading to an outfitter screen that is mostly empty space - which will make finding outfits harder. The only place where having extra categories would really help would be the sales map panels.

**Comment by KiLEdEnNis** [comment]
my problem isn't that "Systems" is that large, it's that it has all sorts of things in it: repairing outfits, expansions, bunk rooms, ramscoops, cooling etc.
no one knows if adding a category would help, so maybe we should organize outfits IN a category in some way?

**Comment by Nescio0** [comment]
> Systems are a relatively large category, but not the largest. 

Personally I'd also favour splitting engines into steering and thrusters, as well as creating a separate category for anti-missile turrets. However, that is beyond the scope of this issue.

My issue is primarily the diversity of the systems category, which is perfectly fine for miscellaneous items, but not really for larger groups of outfits which actually have the same function. Although category size is something to take into account, coherence certainly matters as well. Regenerators have a clear and distinct function, and are quite unlike all other systems.

Furthermore, both suggestions (separating regenerators from systems and cooling from systems) could be implemented independently. Cooling might be a small category, but regenerators would be between secondary weapons and guns; besides, outfitters selling four (Advanced Coalition) or five (Advanced Human) regenerators are not unusual.

> The only place where having extra categories would really help would be the sales map panels.

It would really help there, and slightly help at outfitters (especially if your fleet carries captured (e.g. Korath) systems as cargo), therefore the net benefit is positive. 

Introducing many sorting options and filters would be a great improvement, but also quite complicated to implement; adding a new category is simple and straightforward.

**Comment by Nescio0** [comment]
Yes, I'm aware there exist engines which are both (e.g. X1050) or neither (reverse thruster). “However, that [i.e. discussing splitting engines etc.] is beyond the scope of this issue.”

The primary function of the systems cores is a regenerator; battery and cooling functions are secondary. Likewise, Wanderer cooling engines are listed under engines, because steering or thrust is their primary function; the cooling is nice but not the most important characteristic. Also, the Wanderer sun reactors are primarily power sources, secondarily energy capacitors. Nor do I consider Korath turrets batteries, although they do have a tiny energy capacity.
I have no problems whatsoever with hybrid items combining several functions.

Let's put it differently:
* engines form your ship's movement capabilities
* guns, turrets, secondary weapons, and ammunition together form your ship's offensive capabilities
* regenerators form your ship's passive defensive capabilities
* power deals with your ship's energy management
* cooling systems deal with your ship's heat management

**Comment by Wrzlprnft** [comment]
one thing to consider in your idea: regeneration items do not occupy some sort of exclusive outfit space nor hardpoint

**Comment by Nescio0** [comment]
> one thing to consider in your idea: regeneration items do not occupy some sort of exclusive outfit space nor hardpoint

Nor do ammunition, power, or hand to hand. Also, the reverse thruster occupies weapon capacity, not engine capacity, yet is considered an engine nevertheless. Apparently that's not the criterion for categories.

**Comment by Tadrix** [comment]
I would support splitting _Systems_ into smaller groups. These groups could be _Defense_ (shields, jammers, cloaking), _Sensors_, _Drive systems_ (including ramscoops), _Modifications_ (expansions, bunks, fuel, interference plating), _Cooling systems_.

**Comment by jafdy** [comment]
Sensors would be a minuscule group.

**Comment by Tadrix** [comment]
Agreed, but they don't fit anywhere else. One alternative would be to use _Other systems_ instead of _Modifications_ and throw sensors in there.

> so maybe we should organize outfits IN a category in some way?

How it would be different (in terms of potential problems)? Also, at one time I suggested something similar here at GitHub.

> Remember, if we add a category we can never remove that category.

Let the game auto-populate categories? For example, if I had A (gun), B (gun), C (engine), then game would show [Engines: D; Guns: A, B]. If I then add D (engine) and E (turret), then game shows [Engines: D; Guns: A, B; Turrets: E]. If I then reassign E from turrets to anti-missiles, then there would be [Engines: D; Guns: A, B; Anti-missiles: E]. We could even include category order in data file; listed empty categories wouldn't show, and unlisted categories would be at the bottom of outfitter, sorted alphabetically.

This way mod makers can make their own categories and reorder them, and if we at some point decided to deprecate a category, it wouldn't break anything.

**Comment by TomGoodIdea** [comment]
We now have the "series" feature, which allows setting custom order of outfits through their data definitions, forming something similar to subcategories, without requiring technically separate categories.
See https://github.com/endless-sky/endless-sky/commit/f81af0d53561a02d17175e718e34170e8a39f31e)
